src/safe_apps/serializers.py

```diff
@@ -23,4 +37,10 @@ class Meta:
             "description",
             "chain_ids",
             "provider",
+            "access_control",
         ]
+
+    @staticmethod
+    @swagger_serializer_method(serializer_or_field=AccessControlPolicySerializer)  # type: ignore[misc]
+    def get_access_control(obj: SafeApp) -> ReturnDict:
+        return AccessControlPolicySerializer(obj).data
```

Comment:
On the API level, we are always returning `value` even though we know it's going to be empty if `NO_RESTRICTIONS`. If we always end up exposing both `type` and `value` the clients might want to check both even though `NO_RESTRICTIONS` should always have `"value" : []`.

Suggestion: introduce two serialisers (one for each type). You'd check the type on this function and serialize accordingly:

- if `NO_RESTRICTIONS`

```javascript
"accessControl" : {
  "type": "NO_RESTRICTIONS"
}
```

- if `DOMAIN_ALLOWLIST`

```javascript
"accessControl" : {
  "type": "DOMAIN_ALLOWLIST",
  "value" : [ ... ] // domains
}
```
On the client side, they can then associate each type with a different payload. Eg.: if the client gets `DOMAIN_ALLOWLIST` then it should map to a type that has a collection of domains.